src/views/Resources.vue

```diff
@@ -3,51 +3,98 @@
+            {{
+              sortChoice === "All"
+                ? "All resources table"
+                : `${sortChoice}'s resources table`
+            }}
+          </v-container>
+          <div class="d-flex justify-space-between">
+            <v-menu>
+              <template v-slot:activator="{ props }">
+                <v-btn
+                  variant="outlined"
+                  class="mx-4"
+                  rounded="xs"
+                  :size="isSmAndDown ? 'small' : 'x-large'"
+                  color="red"
+                  v-bind="props"
+                >
+                  Sorting
+                </v-btn>
+              </template>
+              <v-list>
+                <v-list-item
+                  v-for="(item, index) in items"
+                  :key="index"
+                  :value="index"
+                >
+                  <v-list-item-title @click="clickHandler(item.title)">{{
+                    item.title
+                  }}</v-list-item-title>
+                </v-list-item>
+              </v-list>
+            </v-menu>
+            <v-btn
+              class="mx-4"
+              rounded="xs"
+              :size="isSmAndDown ? 'small' : 'x-large'"
+              color="red"
+              to="/resources/add"
+              >Add resource</v-btn
+            >
           </div>
 
-          <resource-table></resource-table>
+          <resource-table :sortChoice="sortChoice"></resource-table>
         </v-card>
       </v-col>
     </v-row>
   </v-container>
 </template>
 
 <script>
-import { onMounted, inject } from "vue";
+import { onMounted, inject, ref, computed } from "vue";
 import ResourceTable from "@/components/Table/ResourceTable.vue";
 import { useStore } from "vuex";
+import { useDisplay } from "vuetify/lib/framework.mjs";
 
 export default {
   components: {
     ResourceTable,
   },
   setup() {
     const store = useStore();
-    const showSnackbar = inject("showSnackbar");
+    const snackbarProvider = inject("snackbarProvider");
+    const mobile = useDisplay();
+    const sortChoice = ref("All");
+
+    const clickHandler = (title) => {
+      sortChoice.value = title;
+    };
 
     onMounted(async () => {
       try {
         await store.dispatch("resources/fetchResources");
       } catch (error) {
-        showSnackbar({
-          message: "Failed to fetch resources.",
-          color: "error",
-          timeout: 4000,
-          location: "top right",
-        });
+        snackbarProvider.showErrorSnackbar("Failed to fetch resources.");
       }
     });
 
-    return {};
+    return {
+      isSmAndDown: computed(() => {
+        return mobile.smAndDown.value;
+      }),
+      clickHandler,
+      sortChoice,
+      items: [
+        { title: "All" },
+        { title: "Pearl" },
+        { title: "LinkingPart" },
+        { title: "Gemstone" },
+        { title: "PreciousMetal" },
+      ],
+    };
```

Comment:
These are all very generic names that don't tell us anything about what we are doing here.
```suggestion
      filterResourcesByType,
      selectedResourceType,
      resourceTypes: "All" ,"Pearl", "LinkingPart", "Gemstone", "PreciousMetal"
      ],
    };
```